requirements.txt

```diff
@@ -1,18 +1,49 @@
 alembic==0.8.5
+asn1crypto==0.24.0
+atomicwrites==1.3.0
+attrs==19.1.0
+certifi==2019.6.16
+cffi==1.12.3
+chardet==3.0.4
+Click==7.0
+clickclick==1.2.2
 connexion==2.2.0
+cryptography==2.7
 Flask==1.0.2
 Flask-Cors==3.0.8
-Flask-Migrate==1.8.0
 flask-marshmallow==0.10.1
+Flask-Migrate==1.8.0
 Flask-Script==2.0.5
 Flask-SQLAlchemy==2.4.0
+idna==2.8
+importlib-metadata==0.20
+inflection==0.3.1
+itsdangerous==1.1.0
+Jinja2==2.10.1
+jsonschema==2.6.0
+Mako==1.1.0
+MarkupSafe==1.1.1
 marshmallow==2.19.2
 marshmallow-enum==1.4.1
 marshmallow-sqlalchemy==0.16.3
+more-itertools==7.2.0
 openapi-spec-validator==0.2.7
-pyopenssl==19.0.0
+packaging==19.1
+pluggy==0.12.0
```

Comment:
Shall we add only the required ones

Reply:
Cleaned requirements.txt

Most of them are transitive dependencies pf pytest,

```
$ pipdeptree -p pytest
pytest==5.1.2
  - atomicwrites [required: >=1.0, installed: 1.3.0]
  - attrs [required: >=17.4.0, installed: 19.1.0]
  - importlib-metadata [required: >=0.12, installed: 0.20]
    - zipp [required: >=0.5, installed: 0.6.0]
      - more-itertools [required: Any, installed: 7.2.0]
  - more-itertools [required: >=4.0.0, installed: 7.2.0]
  - packaging [required: Any, installed: 19.1]
    - attrs [required: Any, installed: 19.1.0]
    - pyparsing [required: >=2.0.2, installed: 2.4.2]
    - six [required: Any, installed: 1.12.0]
  - pluggy [required: >=0.12,<1.0, installed: 0.12.0]
    - importlib-metadata [required: >=0.12, installed: 0.20]
      - zipp [required: >=0.5, installed: 0.6.0]
        - more-itertools [required: Any, installed: 7.2.0]
  - py [required: >=1.5.0, installed: 1.8.0]
  - wcwidth [required: Any, installed: 0.1.7]

```

Reply:
Should we include transitive dependencies ?

Reply:
We can have it either way. If we don't specify the transitive dependencies, pip will pick up the latest of them and it is more clear to see what packages the project depends on. Since we are following this approach, fixed requirements.txt to have only the top level dependency.